packages/base-controller/src/RestrictedControllerMessenger.test.ts

```diff
@@ -502,6 +502,29 @@ describe('RestrictedControllerMessenger', () => {
     ).not.toThrow();
   });
 
+  it('should allow calling an internal action', () => {
+    type CountAction = {
+      type: 'CountController:count';
+      handler: (increment: number) => void;
+    };
+    const controllerMessenger = new ControllerMessenger<CountAction, never>();
+    const restrictedControllerMessenger = controllerMessenger.getRestricted({
+      name: 'CountController',
+      allowedActions: ['CountController:count'],
```

Comment:
From the PR description, it sounds like this isn't necessary, is that true? If so is it worth adding a test which confirms that? Or perhaps I misunderstand.

Reply:
I think you're right. This should probably be:

```suggestion
      allowedActions: [],
```

The test still works!

Reply:
In fact, there are a whole lot of places in this file where this change has to be made: https://github.com/MetaMask/core/pull/2051/commits/dd5bad61c57d8289c2f1fc1c7f795ba10463e149

Reply:
Added a feature for enforcing that no internal actions/events are passed into `RestrictedControllerMessenger`: https://github.com/MetaMask/core/pull/2051/commits/2db90ce9250cdc6d6f5add034d0fb6fab65f88ea

Reply:
Good catch! This was a copy+paste mistake, meant to remove that

Reply:
Good idea to add enforcement! I'll check that PR out soon

Reply:
Fixed in a2956d0e2b68e72beaaa0f07e4a9480b7ee894f7

@MajorLift Your PR looks great but is a bit larger in scope, so I'll leave that to follow this one